src/wolfio.c

```diff
@@ -32,6 +32,15 @@
 
 #ifndef WOLFCRYPT_ONLY
 
+#if defined(HAVE_ERRNO_H) && defined(WOLFSSL_NO_SOCK) && \
+    (defined(USE_WOLFSSL_IO) || defined(HAVE_HTTP_CLIENT))
+    /* error codes are needed for TranslateIoReturnCode() and
+     * wolfIO_TcpConnect() even if defined(WOLFSSL_NO_SOCK), which inhibits
+     * inclusion of errno.h by wolfio.h.
+     */
+    #include <errno.h>
```

Comment:
just for my understanding, why check for `HAVE_ERRNO_H` alone is not enough? what is the downside of including it every time it is includable?

Reply:
Was just narrowing the fix to avoid having to think about collateral damage on other targets/configs.  Chances are it would have been harmless to include `errno.h` whenever `defined(HAVE_ERRNO_H)`.